nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2672,6 +2676,25 @@ public DocumentModel getOrCreateDocument(DocumentModel docModel,
         });
     }
 
+    @Override
+    public DocumentModel newDocumentModel(DocumentRef parentRef, String name, String typeName) {
+        if (parentRef == null && name == null) {
+            return createDocumentModel(typeName);
+        }
+
+        String parentPath = null;
+        if (parentRef != null) {
+            parentPath = resolveReference(parentRef).getPath();
```

Comment:
I would prefer

    String parentPath = parentRef == null ? null : resolveReference(parentRef).getPath();

but up to you